assembly/src/ast/module.rs

```diff
@@ -220,10 +225,16 @@ impl ModuleAst {
         let num_local_procs = source.read_u16()? as usize;
         let local_procs = Deserializable::read_batch_from(source, num_local_procs)?;
 
-        match Self::new(local_procs, reexported_procs, docs) {
-            Err(err) => Err(DeserializationError::UnknownError(err.message().clone())),
-            Ok(res) => Ok(res.with_import_info(import_info)),
+        let mut module_ast = Self::new(local_procs, reexported_procs, docs)
+            .map_err(|err| DeserializationError::UnknownError(err.message().clone()))?
+            .with_import_info(import_info);
+
+        // deserialize source locations if required
+        if options.serialize_source_locations {
+            module_ast.load_source_locations(source)?;
```

Comment:
Similar comment as above: should we get rid of `load_source_locations` as a separate method?